spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -134,6 +134,16 @@
 	 */
 	<T> Iterable<T> query(Query<? extends BaseEntity> query, Class<T> entityClass);
 
+	/**
+	 * Runs given query and applies given function to each entity in the result.
+	 * @param query the query to run.
+	 * @param entityFunc the function to apply to each found entity or key.
+	 * @param <A> the row type of the query.
+	 * @param <T> the type to map each entity or key to.
+	 * @return the mapped entities or keys.
+	 */
+	<A, T> Iterable<T> query(Query<A> query, Function<A, T> entityFunc);
```

Comment:
Can we use `? extends BaseEntity` instead of `A`?
Is it also useful for `Key` queries?

Reply:
No. This function is also used for `Query<Key>`, and `Key` does not extend `BaseEntity`